### PR TITLE
increase coverage of agent to 89.3%

### PR DESCRIPTION
**Pull request by HuayiJin**

### CODE REVIEW

File: test/lib_agentTests.js [hunk 1]
```diff
@@ -24,72 +24,327 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 
 process.env.NODE_ENV = 'test';
-var assert = require('assert'),
+var assert = require('chai').assert,
```
Comment (chuan9) [inline test/lib_agentTests.js]: Pls use original assert lib

Comment (HuayiJin) [reply test/lib_agentTests.js]: ok, should i pr again?

File: test/lib_agentTests.js [hunk 2]
```diff
@@ -24,72 +24,327 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
-            assert(msg, 'msg is null');
-            assert.equal(msg, 'name must be at least 4 characters long, type is missing');
+    it('should fail validate', function(done) {
+        var result
+        var obj = {};
+        var schema = {
+            "type": "boolean"
+        };
+        //console.log(schemaValidation.validate(obj, schema));
+        assert.notEqual(schemaValidation.validate(obj, schema), []);
```
Comment (chuan9) [inline test/lib_agentTests.js]: It will be better that you give the error type instead of not equal

Comment (HuayiJin) [reply test/lib_agentTests.js]: OK, I've rewritten it

File: test/fakecommon.js [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        assert.isString(fullPath, "The fullname is not String");
+    },
+    writeToJson: function() {},
+    getConfig: function() {
+        return Testvalue.config;
+    },
+    saveToUserConfig: function(key, value) {
+        switch (key) {
+        case "connector.ws.host":
+            assert.equal(value, Testvalue.host_value);
```
Comment (chuan9) [inline test/fakecommon.js]: Should not do assertion in fakecommon.js. This should return content if it's reading.

File: test/lib_agentTests.js [hunk 4]
```diff
@@ -25,71 +25,377 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+
+        resetproxy();
+        assert.equal(saveToUserConfigspy.callCount, 4, 'called times error');
+        done();
+    }));
+
+    it('should set and get last actuations pull time', sinonTest(function(done) {
+        var setlastactuationspulltime = configurator_test.__get__("setLastActuationsPullTime");
+        Testvalue.timeTest = 'lasttime';
+        setlastactuationspulltime(Testvalue.timeTest);
```
Comment (chuan9) [inline test/lib_agentTests.js]: Generally speaking, there should be an assertion in every unit test.

**Pull request merged**